sccp/figures/figureThomsonReduce.py

```diff
@@ -16,27 +16,39 @@ def makeFigure():
     rank = 20
     X = import_thomson()
 
-    ax, f = getSetup((12, 12), (2, 3))
+    ax, f = getSetup((12, 12), (1, 1))
     subplotLabel(ax)
 
     # figures for all dataset
     dataX = pf2(X, rank, random_state=1)
 
     factors = [dataX.uns["Pf2_A"], dataX.uns["Pf2_B"], dataX.varm["Pf2_C"]]
-    plotFactors(dataX, ax[0:3], reorder=(0, 2))
+    # plotFactors(dataX, ax[0:3], reorder=(0, 2))
 
-    sampled_data = sc.pp.subsample(X, fraction=0.99, random_state=0, copy=True)
+    # sampled_data = sc.pp.subsample(X, fraction=0.99, random_state=0, copy=True)
+
+    samples = [
```

Comment:
@Nathaniel-github @aarmey I think the problem that arises is when you remove too many cells, some of the genes also get dropped for some reason. Then, during the assertion error of `np.all(np.diff(unique) == 1)` gets called out as False, because some of the indices are missing. Is there a way to reset the indices?

Reply:
@andrewram4287, you're exactly right! I wrote this because the norm calculation implicitly assumes there are no fully empty genes. You can simply comment out the assertion and proceed. Worse case, the R2X is slightly off. However, I'm not completely sure there is a problem at all.